python/ray/_private/runtime_env/runtime_env_agent/runtime_env_agent.py

```diff
@@ -549,12 +617,220 @@ async def GetRuntimeEnvsInfo(self, request, context):
         reply.total = len(runtime_env_states)
         return reply
 
-    async def run(self, server):
-        if server:
+    async def run(self):
+        async def _check_parent():
```

Comment:
This function is copied from agent.py? Can we abstract a common util function?

Reply:
It's not exactly the same, I removed the part that extract and print raylet log